src/cartridges/int_adyen_SFRA/cartridge/adyen/scripts/payments/adyenTerminalApi.js

```diff
@@ -28,6 +28,26 @@ const AdyenHelper = require('*/cartridge/adyen/utils/adyenHelper');
 const AdyenConfigs = require('*/cartridge/adyen/utils/adyenConfigs');
 const constants = require('*/cartridge/adyen/config/constants');
 
+function pspReferenceFromResponse(paymentResponse) {
+  return paymentResponse?.POIData?.POITransactionID?.TransactionID?.split(
+    '.',
+  )[1];
+}
+
+function parsePaymentReceipt(paymentResponse) {
```

Comment:
Will this work if merchant adjusts the payment receipt through BO? 
Should we better rely on extracting this data from `PaymentInstrumentData` which is found under `PaymentResult` on the response?

Reply:
The PaymentInstrumentData has only paymentMethod but not paymentMethodVariant.